### DISCUSSION TITLE
My ImageMagick cannot install webp extension

### DISCUSSION THREAD

**Discussion by znc15** [ImageMagick/ImageMagick#6897] [Help] [answered]
### ImageMagick version

7.1.1-21

### Operating system, version and so on

CentOS 7.9.2009 x86_64

### Description

My ImageMagick cannot install the webp extension. After the ./configure command is executed, it shows that my webp detection is WEBP --with-webp=yes no. But I have installed the libwebp 1.0.2 library, and the cwebp command ran successfully, but it still cannot be installed.

**Comment by fmw42** [comment]
Is its location in your $PATH environment variable so that Imagemagick can find it?   What error show is the config.log file?

**Reply by znc15** [reply]
Thank you for your reply. I have set the reference path for libwebp,this is the config.log file:https://drive.google.com/file/d/1jVits5r_lW3_5tY02nGFrW5hKtLF7gFS/view?usp=drive_link

**Reply by fmw42** [reply]
Your config.log file is full of errors.  I suspect a bad compiler or incompatible compiler.  What was your .config command?

These are the webp errors:

```
configure:33699: checking for libwebp >= 0.4.1
configure:33706: $PKG_CONFIG --exists --print-errors "libwebp >= 0.4.1"
Requested 'libwebp >= 0.4.1' but version of libwebp is 0.3.0
configure:33709: $? = 1
configure:33723: $PKG_CONFIG --exists --print-errors "libwebp >= 0.4.1"
Requested 'libwebp >= 0.4.1' but version of libwebp is 0.3.0
configure:33726: $? = 1
configure:33740: result: no
Requested 'libwebp >= 0.4.1' but version of libwebp is 0.3.0
configure:33770: checking for libwebpmux >= 0.5.0 libwebpdemux >= 0.5.0
configure:33777: $PKG_CONFIG --exists --print-errors "libwebpmux >= 0.5.0 libwebpdemux >= 0.5.0"
Requested 'libwebpmux >= 0.5.0' but version of libwebpmux is 0.3.0
Package libwebpdemux was not found in the pkg-config search path.
Perhaps you should add the directory containing `libwebpdemux.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libwebpdemux' found
configure:33780: $? = 1
configure:33794: $PKG_CONFIG --exists --print-errors "libwebpmux >= 0.5.0 libwebpdemux >= 0.5.0"
Requested 'libwebpmux >= 0.5.0' but version of libwebpmux is 0.3.0
Package libwebpdemux was not found in the pkg-config search path.
Perhaps you should add the directory containing `libwebpdemux.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libwebpdemux' found
configure:33797: $? = 1
configure:33811: result: no
Requested 'libwebpmux >= 0.5.0' but version of libwebpmux is 0.3.0
No package 'libwebpdemux' found

```

Basically, I think it says that it wants webp 0.4.1 and you have installed 0.3.0

Sorry, I am not an expert on reading config.log files.  Perhaps an IM developer can confirm.

**Reply by znc15** [reply]
Thank you, I will try installing a higher version of libwebp later

**Reply by fmw42** [reply]
You should look at all the fatal compiler errors.  Sorry, I do not know what to make of them.

```
gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

configure:5961: $? = 0
configure:5950: gcc -v >&5
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/4.8.5/lto-wrapper
Target: x86_64-redhat-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-linker-hash-style=gnu --enable-languages=c,c++,objc,obj-c++,java,fortran,ada,go,lto --enable-plugin --enable-initfini-array --disable-libgcj --with-isl=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/isl-install --with-cloog=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/cloog-install --enable-gnu-indirect-function --with-tune=generic --with-arch_32=x86-64 --build=x86_64-redhat-linux
Thread model: posix
gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC) 
configure:5961: $? = 0
configure:5950: gcc -V >&5
gcc: error: unrecognized command line option '-V'
gcc: fatal error: no input files
compilation terminated.
configure:5961: $? = 4
configure:5950: gcc -qversion >&5
gcc: error: unrecognized command line option '-qversion'
gcc: fatal error: no input files
compilation terminated.
configure:5961: $? = 4
configure:5950: gcc -version >&5
gcc: error: unrecognized command line option '-version'
gcc: fatal error: no input files
compilation terminated.
configure:5961: $? = 4
configure:5981: checking whether the C compiler works
configure:6003: gcc    conftest.c  >&5
configure:6007: $? = 0
configure:6057: result: yes
configure:6060: checking for C compiler default output file name
configure:6062: result: a.out
configure:6068: checking for suffix of executables
configure:6075: gcc -o conftest    conftest.c  >&5
configure:6079: $? = 0
configure:6102: result: 
configure:6124: checking whether we are cross compiling
configure:6132: gcc -o conftest    conftest.c  >&5
configure:6136: $? = 0
configure:6143: ./conftest
configure:6147: $? = 0
configure:6162: result: no
configure:6167: checking for suffix of object files
configure:6190: gcc -c   conftest.c >&5
configure:6194: $? = 0
configure:6216: result: o
configure:6220: checking whether the compiler supports GNU C
configure:6240: gcc -c   conftest.c >&5
configure:6240: $? = 0
configure:6250: result: yes
configure:6261: checking whether gcc accepts -g
configure:6282: gcc -c -g  conftest.c >&5
configure:6282: $? = 0
configure:6326: result: yes
configure:6346: checking for gcc option to enable C11 features
configure:6361: gcc  -c -g -O2  conftest.c >&5
conftest.c:64:3: error: #error "Compiler does not advertise C99 conformance"
 # error "Compiler does not advertise C99 conformance"
   ^
conftest.c:119:29: error: expected ';', ',' or ')' before 'text'
 test_restrict (ccp restrict text)
                             ^
conftest.c:168:3: error: #error "Compiler does not advertise C11 conformance"
 # error "Compiler does not advertise C11 conformance"
   ^
conftest.c:199:12: error: 'syntax' undeclared here (not in a function)
 #define u8 syntax error!
            ^
conftest.c:200:29: note: in expansion of macro 'u8'
 char const utf8_literal[] = u8"happens to be ASCII" "another string";
                             ^
conftest.c:199:19: error: expected ',' or ';' before 'error'
 #define u8 syntax error!
                   ^
conftest.c:200:29: note: in expansion of macro 'u8'
 char const utf8_literal[] = u8"happens to be ASCII" "another string";
                             ^
conftest.c: In function 'main':
conftest.c:233:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'newvar'
   char *restrict newvar = "Another string";
                  ^
conftest.c:233:18: error: 'newvar' undeclared (first use in this function)
conftest.c:233:18: note: each undeclared identifier is reported only once for each function it appears in
conftest.c:243:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < ia->datasize; ++i)
   ^
conftest.c:243:3: note: use option -std=c99 or -std=gnu99 to compile your code
configure:6361: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "ImageMagick"
| #define PACKAGE_TARNAME "ImageMagick"
| #define PACKAGE_VERSION "7.1.1-21"
| #define PACKAGE_STRING "ImageMagick 7.1.1-21"
| #define PACKAGE_BUGREPORT "https://github.com/ImageMagick/ImageMagick/issues"
| #define PACKAGE_URL "https://imagemagick.org"
| #define MAGICK_TARGET_CPU x86_64
| #define MAGICK_TARGET_VENDOR pc
| #define MAGICK_TARGET_OS linux-gnu
| /* end confdefs.h.  */
| 
| /* Does the compiler advertise C89 conformance?
|    Do not test the value of __STDC__, because some compilers set it to 0
|    while being otherwise adequately conformant. */
| #if !defined __STDC__
| # error "Compiler does not advertise C89 conformance"
| #endif

```

**Reply by znc15** [reply]
It's okay. I'd like to thank you very much if you can help analyze the error report.

**Reply by fmw42** [reply]
Sorry, I cannot help further.  One of the Imagemagick developers might need to comment on this.

**Comment by snibgo** [comment]
`config` has to discover what facilities the compiler and standard libraries have. It does this by writing many small programs and compiling them, or by executing simple test commands. When one of those tests "fails", then that facility is not available. For example, "`gcc -v`" works but "`gcc -version`" doesn't. Failing those tests is not a problem. We can see towards the bottom of config.log that it can successfully build IM.

I do notice that the gcc version is:
```
gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC)
```

For comparison, on my Windows plus Cygwin laptop, my gcc is:
```
gcc version 11.3.0 (GCC)
```
So znc15's compiler seems to be old. This doesn't create a fatal problem, but an upgrade may be helpful, eg for performance.

fmw42's comments about "`checking for libwebp >= 0.4.1`" etc are correct. Similarly, config could not find OpenEXR.

**Reply by znc15** [reply]
I've updated libwep, but it still doesn't work
[root@LittlesheepUAU ImageMagick-7.1.1-21]# cwebp -version
1.3.2
libsharpyuv: 0.2.1
[root@LittlesheepUAU ImageMagick-7.1.1-21]# dwebp -version

**Reply by znc15** [reply]
But the log shows that I still want >=0.4.1

**Comment by fmw42** [comment]
As I said above it is looking for version 0.4.1.

**Reply by znc15** [reply]
But isn’t it also okay if it’s greater than 0.4.1?
The config.log file https://drive.google.com/file/d/1iqnF_iEm8LaCS2D0unePs5eh-zhyYLIO/view?usp=drive_link

**Reply by fmw42** [reply]
Yes, should be. It works fine for me with webp at 1.3.0.  Do you now have any errors about webp in your config.log? If so, fix them and try install IM again.

**Reply by fmw42** [reply]
How do you know your latest install is not working?  Does webp show in the list of delegates from

`magick -version`

**Reply by znc15** [reply]
[root@LittlesheepUAU ~]# magick -version
Version: ImageMagick 7.1.1-21 Q16-HDRI x86_64 21667 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(3.1) 
Delegates (built-in): bzlib fontconfig freetype jng jpeg lzma png tiff x xml zip zlib
Compiler: gcc (4.8)
[root@LittlesheepUAU ~]# 

I installed libwebp from Google official website, cwebp can already be used, but libwebp cannot be found, and the 'rpm -qa | grep libwebp' command has no output.

**Comment by snibgo** [comment]
In your first config.log, you had:
```
configure:33706: $PKG_CONFIG --exists --print-errors "libwebp >= 0.4.1"
Requested 'libwebp >= 0.4.1' but version of libwebp is 0.3.0
configure:33709: $? = 1
```
So it found libwebp, but an old version.

In your latest config.log, you have:
```
configure:33706: $PKG_CONFIG --exists --print-errors "libwebp >= 0.4.1"
Package libwebp was not found in the pkg-config search path.
Perhaps you should add the directory containing `libwebp.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libwebp' found
```
Oops. Now it can't find libwebp at all.

**Reply by znc15** [reply]
The libwebp version installed on Centos7's yum is too old, and manual installation is not useful

**Comment by fmw42** [comment, marked as answer]
Do you have pkg-config installed? If not install it

If so create an entry for it in your .bash_profile or .zprofile or both in your home directory 

Add this to your .bash_profile

export PKG_CONFIG_PATH="/usr/local/bin/pkg-config"

or wherever you have pkg-config installed.

If doing just that does not work, then add the path to where libwebp is installed.  

Or perhaps just add the path to libwebp in your $PATH env variable in the same files or in both places?

**Reply by znc15** [reply]
Thank you very much, webp has appeared

**Reply by znc15** [reply]
WEBP --with-webp=yes yes

**Reply by znc15** [reply]
Then run ‘convert -list format’ after compilation, and the error message ‘convert: error while loading shared libraries: libsharpyuv.so.0: cannot open shared object file: No such file or directory‘

**Reply by znc15** [reply]
It's okay, I have successfully installed it, thank you for your efforts
![image](https://github.com/ImageMagick/ImageMagick/assets/64775575/16e201d7-a5f4-4901-a613-113485d8261a)
